### PR TITLE
build proto files without git submodule

### LINKED ISSUE #28: (possible?) replace Git submodules for retrieving proto files with explicit cargo.toml dependency instead

### PR DESCRIPTION
**Pull request by benjaminbollen**
resolves #28 

@vadim-frolov I started with an experiment to download the protofiles from raw github content, with the aim to remove git submodule from `cargo build` step (ideally this would be resolved over a CID :) )

could you have a look if you like the idea?

open todos:
- [ ] actually prost build the proto to .rs
- [ ] update the imports in code
- [ ] should this be build.rs import a lib, because the imports need to be build multiple times?

### CODE REVIEW

File: substreams-sink/Cargo.toml [hunk 1]
```diff
@@ -12,4 +12,7 @@ tokio = { version = "1.0", features = ["macros", "rt-multi-thread"] }
 tokio-stream = { version = "0.1", features = ["net"] }
 
 [build-dependencies]
-tonic-build = "0.8.4"
\ No newline at end of file
+tonic-build = "0.8.4"
+curl = "0.4.44"
+prost-build = "0.11.7"
+tempdir = "0.3.7"
```
Comment (benjaminbollen) [inline substreams-sink/Cargo.toml:18]: this should be removed

**Review by jorgeantonio21 — COMMENTED** [review]
It is looking good to me :)

**Comment by vadim-frolov** [comment]
Good to get rid of submodules. I would try to minimize the refactoring and continue using autogenerated tonic code if you don't mind. I came across git-download crate when searching for alternatives. What do you think #64 ?

**Comment by benjaminbollen** [comment]
superceded by #64

**Pull request closed without merging**